### ISSUE TITLE
probe returns ready before CAYW can actually be called

### ISSUE THREAD

**Issue by AmomentOfMusic** [retorquere/zotero-better-bibtex#1191]
In my powershell script to call up the CAYW picker, I am trying to implement a feature in which the script checks if Zotero is open, and if not, launchs the application for you and then continues with calling the picker. In order to do so, I am using the BBT probe to wait until it returns ready before proceeding.

Issue: Probe returns ready before I can actually call the picker.

In order to illustrate my issue, here is a test script I've been playing with:
```
function Test-Site {
    param($URL)
    trap{
        Start-Sleep -Seconds 2
        Write-Host "waiting"
    }
    $webclient = New-Object Net.WebClient
    # This is the main call
    $test = $webclient.DownloadString($URL)
    if ($test -ne "ready")
    {return "waiting"}
    Else {return "$test"}
     }

Do {
    $probe = Invoke-Expression "Test-Site 'http://localhost:23119/better-bibtex/cayw?probe=true'"
   }

Until ($probe -eq "ready")

echo $probe
curl "http://localhost:23119/better-bibtex/cayw?format=scannable-cite" -usebasicparsing
```
When I run this script and then proceed to launch zotero, the probe seems to return "ready" when the BBT plugin initilizes (i.e. when the status window first appears) rather then when it is fully loaded. 

This is confirmed by my results: 
```
waiting
Exception calling "DownloadString" with "1" argument(s): "Unable to connect to the remote server"
At line:10 char:5
+     $test = $webclient.DownloadString($URL)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException
 
ready
curl : The remote server returned an error: (500) Internal Server Error.
At line:23 char:1
+ curl "http://localhost:23119/better-bibtex/cayw?format=scannable-cite ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
```

My code loops through my the test script once until the probe returns ready. However, it then fails to call the picker, saying that the server is not yet available. 

When I run the script when Zotero is already up and running, it calls the picker without issue. I can also hard code the script to pause for about 5 seconds after the probe returns "ready" which gives me successful results, but I would prefer a more elegant solution, as I can't obviously guarantee how long it will take for BBT to load fully.

In short, would it be possible to change the behavior so that probe only returns ready once the plugin is fully loaded, rather than when it initializes?

**Comment by blip-bloop** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.1.83.4050](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.1.83.4050.xpi) ("cayw: report starting until startup completed")

Install in Zotero by downloading [test build 5.1.83.4050](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.1.83.4050.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by retorquere** [comment]
Give 4050 a go.

**Comment by AmomentOfMusic** [comment]
Woohoo! Problem fixed. Thanks for the quick turnaround.

**Comment by retorquere** [comment]
Cool -- new release is scheduled, should be out in 10 minutes or so.
